internal/runners/builds/builds.go

```diff
@@ -0,0 +1,202 @@
+package builds
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/analytics"
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/runbits/rationalize"
+	"github.com/ActiveState/cli/internal/runbits/runtime"
+	bpModel "github.com/ActiveState/cli/pkg/platform/api/buildplanner/model"
+	auth "github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildplan"
+	"github.com/ActiveState/cli/pkg/platform/runtime/store"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+type primeable interface {
+	primer.Outputer
+	primer.Auther
+	primer.Projecter
+	primer.SvcModeler
+	primer.Configurer
+	primer.Analyticer
+}
+
+type Params struct {
+	All bool
+}
+
+type Builds struct {
+	out       output.Outputer
+	project   *project.Project
+	analytics analytics.Dispatcher
+	svcModel  *model.SvcModel
+	auth      *auth.Auth
+	config    *config.Instance
+}
+
+type StructuredOutput struct {
+	Platforms []*structuredPlatform `json:"platforms"`
+}
+
+func (o *StructuredOutput) MarshalStructured(output.Format) interface{} {
+	return o
+}
+
+type structuredPlatform struct {
+	ID       string             `json:"id"`
+	Name     string             `json:"name"`
+	Builds   []*structuredBuild `json:"builds"`
+	Packages []*structuredBuild `json:"packages"`
+}
+
+type structuredBuild struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+	URL  string `json:"url"`
+}
+
+func New(p primeable) *Builds {
+	return &Builds{
+		out:       p.Output(),
+		project:   p.Project(),
+		auth:      p.Auth(),
+		svcModel:  p.SvcModel(),
+		config:    p.Config(),
+		analytics: p.Analytics(),
+	}
+}
+
+func rationalizeError(err *error) {
+	switch {
+	case err == nil:
+		return
+	case errors.Is(*err, rationalize.ErrNoProject):
+		*err = errs.WrapUserFacing(*err,
+			locale.Tr("err_no_project"),
+			errs.SetInput())
+	}
+}
+
+func (b *Builds) Run(params *Params) (rerr error) {
+	defer rationalizeError(&rerr)
+
+	if b.project == nil {
+		return rationalize.ErrNoProject
+	}
+
+	// We don't use the runtime returned here, because builds needs more advanced runtime info, but we still want to call
+	// it in case our runtime isn't up-to date.
+	_, err := runtime.NewFromProject(b.project, target.TriggerBuilds, b.analytics, b.svcModel, b.out, b.auth, b.config)
+	if err != nil {
+		return locale.WrapInputError(err, "err_refresh_runtime_new", "Could not update runtime for this project.")
+	}
+
+	runtimeStore := store.New(target.NewProjectTarget(b.project, nil, target.TriggerBuilds).Dir())
+	bp, err := runtimeStore.BuildPlan()
```

Comment:
The `store` is part of the initialized runtime. The runtime type has methods for resolved artifacts, we could add a method for to give us the build plan and use the runtime that we aren't assigning above.